src/main/java/com/capstone/service/impl/LearnerRoadmapServiceImpl.java

```diff
@@ -301,4 +308,47 @@ private boolean isPreviousAtomCompleted(UUID learnerId, UUID capsuleId, Integer
             })
             .orElse(false);
     }
+
+    private void assignLearnerToMentor(RoadmapRequestDto roadmapRequestDto){
+        MentorSnapshot mentor = selectMentorToAssign(roadmapRequestDto);
+        UserSnapshot learner = userSnapshotRepository.findByUserId(roadmapRequestDto.getLearnerId())
+                .orElseThrow( () -> new UserNotFoundException("A learner provided doesn't exist")
+                );
```

Comment:
Also, these add unnecessary performance cost to the operation, which can be avoided by using entity references instead.
You may retain this for the moment, but always consider best practices when dealing with DB querying and performance trade-offs.

For example:
```Java
// get references without querying
MentorSnapshot mentorRef = entityManager.getReference(MentorSnapshot.class, mentorId);
UserSnapshot learnerRef = entityManager.getReference(UserSnapshot.class, roadmapRequestDto.getLearnerId());
```